homeassistant/components/tibber/__init__.py

```diff
@@ -40,9 +43,18 @@
 
     try:
         await tibber_connection.update_info()
-    except asyncio.TimeoutError as err:
-        _LOGGER.error("Timeout connecting to Tibber: %s ", err)
-        return False
+    except asyncio.TimeoutError:
+        _LOGGER.warning(
+            "Timeout connecting to Tibber. Will retry in %ss", DEFAULT_RETRY
+        )
+
+        async def retry_setup(now):
+            """Retry setup if a timeout happens on Tibber API."""
+            await async_setup(hass, config)
+
+        async_call_later(hass, DEFAULT_RETRY, retry_setup)
```

Comment:
Should we add a back off interval? We could have a simple algorithm something like this:
https://github.com/home-assistant/core/blob/dd91b51435463ca06847d6290cdd8db091bbce78/homeassistant/components/netgear_lte/__init__.py#L336

Reply:
Make sense.

Should I add a delay parameter to `async_setup` ?

Reply:
Yeah, an optional parameter could work. Or save it in hass.data.